pkg/sql/ttl/ttljob/ttljob_processor.go

```diff
@@ -224,18 +225,25 @@ func (ttl *ttlProcessor) work(ctx context.Context) error {
 		return err
 	}
 
-	job, err := serverCfg.JobRegistry.LoadJob(ctx, ttlSpec.JobID)
-	if err != nil {
-		return err
-	}
-	return db.Txn(ctx, func(_ context.Context, txn *kv.Txn) error {
-		return job.Update(ctx, txn, func(_ *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
+	jobID := ttlSpec.JobID
+	return serverCfg.JobRegistry.UpdateJobWithTxn(
+		ctx,
+		jobID,
+		nil,  /* txn */
+		true, /* useReadLock */
+		func(_ *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
 			progress := md.Progress
-			progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount += rowCount
+			existingRowCount := progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount
+			progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount += processorRowCount
```

Comment:
I don't think this needs to use `atomic.LoadInt64` because `processorRowCount` is no longer being modified after `group.Wait()` finishes.